test/Microsoft.AspNet.BasicMiddleware.Tests/HttpMethodOverrideMiddleware.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.TestHost;
+using Xunit;
+
+namespace Microsoft.AspNet.BasicMiddleware.Tests
+{
+    public class HttpMethodOverrideMiddlewareTest
+    {
+        [Fact]
+        public async Task HttpMethodOverride_HeaderAvailable()
+        {
+            var server = TestServer.Create(app =>
+            {
+                app.UseHttpMethodOverrideMiddleware();
+                app.Run(context =>
+                {
+                    Assert.Equal("Put", context.Request.Headers.Get("X-Http-Method-Override"));
```

Comment:
1. You forgot to check if this code is actually executed before ending the test
2. You should check against `context.Request.Method` not the header because that will always be set since you did that a few lines below.